mindsdb/integrations/kafka/kafkadb.py

```diff
@@ -0,0 +1,172 @@
+        if config_record is None:
+            return False
+        integrations = json.loads(config_record.data)["integrations"]
+        if self.name not in integrations:
+            return False
+        return True
+
+    def setup(self):
+        """Launches streams stored in db and
+        Launches a worker in separate thread which waits
+        data from control stream and creates a particular Streams."""
+
+        self.start_stored_streams()
+        # launch worker which reads control steam and spawn streams
+        self.start()
+
+    def start(self):
+        Thread(target=Kafka.work, args=(self, )).start()
+
+    def stop_deleted_streams(self):
+        existed_streams = session.query(Stream).filter_by(company_id=self.company_id, integration=self.name)
+        actual_streams = [x.name for x in existed_streams]
+
+
+        for stream in self.streams.copy():
+            if stream not in actual_streams:
+                # this stream is still running but it has been deleted from database.
+                # need to stop it.
+                self.streams[stream].set()
+                del self.streams[stream]
+
+    def start_stored_streams(self):
+        existed_streams = session.query(Stream).filter_by(company_id=self.company_id, integration=self.name)
+
+        for stream in existed_streams:
+            to_launch = self.get_stream_from_db(stream)
+            if stream.name not in self.streams:
+                params = {"integration": stream.integration,
+                          "predictor": stream.predictor,
+                          "stream_in": stream.stream_in,
+                          "stream_out": stream.stream_out,
+                          "type": stream._type}
+
+                log.error(f"Integration {self.name} - launching from db : {params}")
+                to_launch.start()
+                self.streams[stream.name] = to_launch.stop_event
+
+
+    def delete_stream(self, predictor):
+        """Deletes stream from database and stops it work by
+        setting up a special threading.Event flag."""
+        stream_name = f"{self.name}_{predictor}"
+        log.debug(f"Integration {self.name}: deleting {stream_name}")
+        session.query(Stream).filter_by(company_id=self.company_id, integration=self.name, name=stream_name).delete()
+        session.commit()
+        if stream_name in self.streams:
+            self.streams[stream_name].set()
+            del self.streams[stream_name]
+
+    def work(self):
+        self.consumer = kafka.KafkaConsumer(bootstrap_servers=f"{self.host}:{self.port}",
+                                            consumer_timeout_ms=1000)
+
+        self.consumer.subscribe([self.control_topic_name])
+        log.error(f"Integration {self.name}: subscribed  to {self.control_topic_name} kafka topic")
+        while not self.stop_event.wait(0.5):
+            try:
+                # break if no record about this integration has found in db
```

Comment:
If this comment is necessary, than the method should be called `integration_in_db`  or something similar, if the method is or will be checking for more than db presence, then remove this comment

Reply:
Renamed to `exist_in_db`